core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsBillingProjectComponent.scala

```diff
@@ -79,6 +79,10 @@ trait RawlsBillingProjectComponent {
       findBillingProjectsByBillingAccount(billingAccount).map(_.invalidBillingAccount).update(isInvalid)
     }
 
+    def updateBillingProjectBillingAccountValidity(projectName: RawlsBillingProjectName, isInvalid: Boolean): WriteAction[Int] = {
+      findBillingProjectByName(projectName).map(_.invalidBillingAccount).update(isInvalid)
+    }
+
     def updateBillingAccount(projectName: RawlsBillingProjectName, billingAccount: Option[RawlsBillingAccountName]): WriteAction[Int] = {
       findBillingProjectByName(projectName).map(_.billingAccount).update(billingAccount.map(_.value))
```

Comment:
Maybe you can find the project once then perform the updates to it?

Reply:
Yeah, I could just fold it into `updateBillingAccount`? I thought it might be cleaner to do separations of concerns, but I think they are maybe just fundamentally linked.